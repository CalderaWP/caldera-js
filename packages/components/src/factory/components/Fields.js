import React, { Fragment } from 'react';
import { Field } from './Field';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { collectFieldValues } from './collectFieldValues';
export const Fields = ({ fields, onChange, onBlur, className }) => (
	<div className={classNames('caldera-fields-wrapper', className)}>
		{fields.map(field => (
			<Field
				key={field.fieldId}
				field={field}
				onChange={newValue => {
					onChange({
						...collectFieldValues(fields),
						[field.fieldId]: newValue
					});
				}}
				onBlur={onBlur}
			/>
		))}
	</div>
);

Fields.propTypes = {
	field: PropTypes.shape({
		fieldId: PropTypes.string.isRequired
		//input propTypes?
	}),
	onChange: PropTypes.func.isRequired,
	onBlur: PropTypes.func,
	className: PropTypes.string
};
