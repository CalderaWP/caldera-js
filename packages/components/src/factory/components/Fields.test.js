import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	emailField
} from '../fields.fixtures';

import { Fields } from './Fields';

describe('Fields component', () => {
	let onChange;
	let onBlur;
	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});
	it('Creates a text and checkbox field field', () => {
		const fields = [textField, checkboxField];
		const component = renderer.create(
			<Fields fields={fields} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Creates all five fields', () => {
		const fields = [
			checkboxFieldset,
			selectField,
			checkboxField,
			numberField,
			textField,
			emailField
		];
		const component = renderer.create(
			<Fields fields={fields} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});

describe('Fields component change handling', () => {
	let onChange = jest.fn();
	let onBlur;
	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	const textInitalValue = 'TEXTS';
	const fields = [
		{ ...selectField, value: 1 },
		{ ...textField, value: textInitalValue }
	];

	const component = mount(
		<Fields fields={fields} onChange={onChange} onBlur={onBlur} />
	);

	const newSelectValue = 2;
	const event = { target: { value: newSelectValue } };
	component.find('select').simulate('change', event);
	expect(onChange.mock.calls[0][0]).toEqual({
		[textField.fieldId]: textInitalValue,
		[selectField.fieldId]: newSelectValue
	});
});
