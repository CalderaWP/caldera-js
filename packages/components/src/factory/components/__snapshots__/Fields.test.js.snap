// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields component Creates a text and checkbox field field 1`] = `
<div
  className="caldera-fields-wrapper"
>
  <div
    className="components-base-control caldera-field caldera-field-text"
  >
    <div
      className="components-base-control__field"
    >
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-1"
      >
        First Name
      </label>
      <input
        aria-describedby="inspector-text-control-1__help"
        className="components-text-control__input"
        id="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value="roy"
      />
    </div>
    <p
      className="components-base-control__help"
      id="inspector-text-control-1__help"
    >
      your first name
    </p>
  </div>
  <div
    className="components-base-control"
  >
    <div
      className="components-base-control__field"
    >
      <input
        aria-describedby="inspector-checkbox-control-0__help"
        checked={true}
        className="components-checkbox-control__input"
        id="checkboxFieldLabel"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="1"
      />
      <label
        className="components-checkbox-control__label"
        htmlFor="inspector-checkbox-control-0"
      >
        Checkbox Labe
      </label>
    </div>
    <p
      className="components-base-control__help"
      id="inspector-checkbox-control-0__help"
    >
      Checkbox field description
    </p>
  </div>
</div>
`;

exports[`Fields component Creates all five fields 1`] = `
<div
  className="caldera-fields-wrapper"
>
  <fieldset
    className="caldera-fieldset caldera-fieldset-checkboxes"
  >
    <legend>
      Checkbox Field Set Label
    </legend>
    <div
      className="components-base-control"
    >
      <div
        className="components-base-control__field"
      >
        <input
          aria-describedby={undefined}
          checked={true}
          className="components-checkbox-control__input"
          id="opt-1"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
          value="1"
        />
        <label
          className="components-checkbox-control__label"
          htmlFor="inspector-checkbox-control-1"
        >
          One
        </label>
      </div>
    </div>
    <div
      className="components-base-control"
    >
      <div
        className="components-base-control__field"
      >
        <input
          aria-describedby="inspector-checkbox-control-2__help"
          checked={true}
          className="components-checkbox-control__input"
          id="opt-2"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
          value="1"
        />
        <label
          className="components-checkbox-control__label"
          htmlFor="inspector-checkbox-control-2"
        >
          Two
        </label>
      </div>
      <p
        className="components-base-control__help"
        id="inspector-checkbox-control-2__help"
      >
        The Second Option
      </p>
    </div>
    <div
      className="components-base-control"
    >
      <div
        className="components-base-control__field"
      >
        <input
          aria-describedby={undefined}
          checked={true}
          className="components-checkbox-control__input"
          id="opt-3"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
          value="1"
        />
        <label
          className="components-checkbox-control__label"
          htmlFor="inspector-checkbox-control-3"
        >
          Three
        </label>
      </div>
    </div>
  </fieldset>
  <div
    className="components-base-control caldera-field caldera-field-select"
  >
    <div
      className="components-base-control__field"
    >
      <label
        className="components-base-control__label"
        htmlFor="inspector-select-control-1"
      >
        Select Field Label
      </label>
      <select
        aria-describedby="inspector-select-control-1__help"
        className="components-select-control__input"
        id="selectFieldId"
        multiple={false}
        onChange={[Function]}
        value={1}
      >
        <option
          value={0}
        >
          Zero
        </option>
        <option
          value={1}
        >
          One
        </option>
        <option
          value={2}
        >
          Two
        </option>
      </select>
    </div>
    <p
      className="components-base-control__help"
      id="inspector-select-control-1__help"
    >
      Select field description
    </p>
  </div>
  <div
    className="components-base-control"
  >
    <div
      className="components-base-control__field"
    >
      <input
        aria-describedby="inspector-checkbox-control-4__help"
        checked={true}
        className="components-checkbox-control__input"
        id="checkboxFieldLabel"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="1"
      />
      <label
        className="components-checkbox-control__label"
        htmlFor="inspector-checkbox-control-4"
      >
        Checkbox Labe
      </label>
    </div>
    <p
      className="components-base-control__help"
      id="inspector-checkbox-control-4__help"
    >
      Checkbox field description
    </p>
  </div>
  <div
    className="components-base-control caldera-field caldera-field-number"
  >
    <div
      className="components-base-control__field"
    >
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-2"
      >
        Number Field Label
      </label>
      <input
        aria-describedby="inspector-text-control-2__help"
        className="components-text-control__input"
        id="numberFieldId"
        max={10}
        min={1}
        onBlur={[Function]}
        onChange={[Function]}
        placeholder={undefined}
        type="number"
        value={3}
      />
    </div>
    <p
      className="components-base-control__help"
      id="inspector-text-control-2__help"
    >
      Number field description
    </p>
  </div>
  <div
    className="components-base-control caldera-field caldera-field-text"
  >
    <div
      className="components-base-control__field"
    >
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-3"
      >
        First Name
      </label>
      <input
        aria-describedby="inspector-text-control-3__help"
        className="components-text-control__input"
        id="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value="roy"
      />
    </div>
    <p
      className="components-base-control__help"
      id="inspector-text-control-3__help"
    >
      your first name
    </p>
  </div>
  <div
    className="components-base-control caldera-field caldera-field-email"
  >
    <div
      className="components-base-control__field"
    >
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-4"
      >
        Email Field Label
      </label>
      <input
        aria-describedby="inspector-text-control-4__help"
        className="components-text-control__input"
        id="emailFieldId"
        multiple={true}
        onBlur={[Function]}
        onChange={[Function]}
        placeholder={undefined}
        type="email"
        value="emai@email.com"
      />
    </div>
    <p
      className="components-base-control__help"
      id="inspector-text-control-4__help"
    >
      Email field description
    </p>
  </div>
</div>
`;
